modules/service/src/main/scala/lucuma/odb/graphql/mapping/GmosNorthLongSlitMapping.scala

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+
+package mapping
+
+import cats.syntax.all.*
+import edu.gemini.grackle.Mapping
+import edu.gemini.grackle.Path
+import edu.gemini.grackle.Predicate
+import edu.gemini.grackle.Predicate.*
+import edu.gemini.grackle.Query
+import edu.gemini.grackle.Query.*
+import edu.gemini.grackle.Result
+import edu.gemini.grackle.TypeRef
+import edu.gemini.grackle.skunk.SkunkMapping
+import lucuma.core.enums.GmosAmpGain
+import lucuma.core.enums.GmosAmpReadMode
+import lucuma.core.enums.GmosRoi
+import lucuma.core.enums.GmosYBinning
+import lucuma.core.model.Observation
+import lucuma.odb.graphql.predicate.Predicates
+
+import binding.*
+import input.*
+import table.*
+
+trait GmosNorthLongSlitMapping[F[_]]
+  extends GmosNorthLongSlitTable[F] { this: SkunkMapping[F] =>
+
+  private val DefaultAmpReadMode: GmosAmpReadMode =
+    GmosAmpReadMode.Slow
+    
+  private val DefaultAmpGain: GmosAmpGain =
+    GmosAmpGain.Low
+    
+  private val DefaultRoi: GmosRoi =
+    GmosRoi.FullFrame  
+
+  private val DefaultYBinning: GmosYBinning =
+    GmosYBinning.Two
+
+  lazy val GmosNorthLongSlitMapping: ObjectMapping =
+    ObjectMapping(
+      tpe = GmosNorthLongSlitType,
+      fieldMappings = List(
+        SqlField("observationId", GmosNorthLongSlitTable.ObservationId, key = true, hidden = true),
+
+        SqlField("grating", GmosNorthLongSlitTable.Grating),
+        SqlField("filter",  GmosNorthLongSlitTable.Filter),
+        SqlField("fpu",     GmosNorthLongSlitTable.Fpu),
+        SqlObject("centralWavelength"),
+    
+        // xBin
+//        FieldRef[Option[GmosYBinning]]("explicitXBin").as("yBin", _.getOrElse(defaultXBin)),
+
+        // defaultXBin
+        // TODO: a function of FPU, target SourceProfile, and ImageQuality
```

Comment:
The default x-binning calculation involves looking up the FPU (which I think I can get with a simple `FieldRef`) along with the target's `SourceProfile` and `ImageQuality`, which are used to get the effective slit width.  I don't know how to handle the latter two.

Reply:
I talked to Miles and I have an idea of how to do this. If you want to punt for now and leave this field unmapped (or mapped to a constant) I can come back and fill it in.

Reply:
That sounds great, thanks.  I'll map it to a constant for now and add the offsets.  Then I should probably stop and merge this checkpoint before it gets any bigger!